keps/sig-release/2853-kubernetes-kubernetes-repository-branch-rename/README.md

```diff
@@ -0,0 +1,760 @@
+-->
+
+In the Kubernetes community and other communities, there is an effort to make all the language and terms more inclusive. One small step to make that is to change the current branch name to one that is more inclusive and does not harmful and exclusionary.
+
+Our biggest repository is [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes), and this has a lot of open Pull Requests and several other forks, clones that use that for downstream signal or a custom build.
+
+Changing the branch name might cause issues for downstream consumers, our contributors, and the community in general.
+
+For that, we are creating this KEP to have a plan to address the issues and communication.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+The main goal is to be more inclusive using better naming.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+This applies only to the main Kubernetes GitHub repository [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes).
+Other repositories will follow a similar approach but at their timeframe.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+- The change can be applied in the middle of December 2021. This is a good time because we will have less activity due to the Christmas and new year celebrations
+- Announce the change in the kubernetes-dev and kubernetes-announcement mailing list and maybe in a blog post. As well as use Twitter and other social media to announce.
+- Change all open Pull Requests to the `Draft mode` with that when we rename the branch, the tests will not be triggered and will avoid a massive queue in the CI infrastructure.
+- Update all Prow jobs that use `kubernetes/kubernetes` and use the `master` branch to listen to the `main` branch as well
+- Follow the guide on https://github.com/kubernetes/community/blob/master/github-management/default-branch-migration.md to rename the repository branch name.
+- After the branch rename, announce that in the mailing list
+- Update all Prow jobs that use `kubernetes/kubernetes` and to remove the `master` branch
+- Convert on demand the open Pull Requests from the `Draft mode` to `Review ready mode`
+
+
+References:
+- https://github.com/kubernetes/org/issues/2222
+- https://github.com/kubernetes/enhancements/issues/2853
+- https://github.com/kubernetes/kubernetes/issues/105601
+- https://github.com/kubernetes/community/blob/master/github-management/default-branch-migration.md
+
+- https://github.com/github/renaming
+
```

Comment:
We should additionally gather input from downstream consumers how it affects their workflow. Let's do a survey for this and add the questions as part of this KEP. For example:

- On which companies behalf do you submit the response?
- Would a k/k branch rename affect your downstream workflow? If yes, how?
- How much time would you need in advance before the migration happens?

Reply:
added the initial questions